File: R/utils_db.R
library(DBI)
library(RSQLite)

sqlite_path <- function(year_qtr) {
  file.path(cfg$sqlite_dir, paste0(year_qtr, ".sqlite"))
}

connect_sqlite <- function(year_qtr) {
  dbConnect(SQLite(), sqlite_path(year_qtr))
}

disconnect_sqlite <- function(con) {
  if (!is.null(con)) dbDisconnect(con)
}

File: python/py/utils_db.py
from __future__ import annotations

import sqlite3
from pathlib import Path

from py.config import cfg


def sqlite_path(year_qtr: str | int) -> Path:
    return cfg.sqlite_dir / f"{year_qtr}.sqlite"


def connect_sqlite(year_qtr: str | int) -> sqlite3.Connection:
    return sqlite3.connect(sqlite_path(year_qtr))


def disconnect_sqlite(con: sqlite3.Connection | None) -> None:
    if con is not None:
        con.close()
